### PR TITLE
Allow `insertFrom` for `ColumnFunction`

### LINKED ISSUE #7462: Error. Cannot get value from Function with ARRAY JOIN over lambda + limit by
```
DROP TABLE TESTTABLE;
CREATE TABLE TESTTABLE (
  _id UInt64,  pt String, attr_list Array(String)
) ENGINE = MergeTree() PARTITION BY (pt) ORDER BY  tuple();

insert into TESTTABLE  values (0,'0',['1']),(1,'1',['1']);

SELECT attr, _id, arrayFilter(x -> (x IN (select '1')), attr_list) z
FROM TESTTABLE ARRAY JOIN z AS attr ORDER BY _id LIMIT 3 BY attr

→ Progress: 0.00 rows, 0.00 B (0.00 rows/s., 0.00 B/s.) Received exception from server (version 19.15.2):
Code: 48. DB::Exception: Received from localhost:9000. DB::Exception: Cannot get value from Function.

```
 19.15.2.2 (official build).

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow some queries with sorting, LIMIT BY, ARRAY JOIN and lambda functions. This closes #7462.

### CODE REVIEW

File: src/Columns/ColumnFunction.cpp [hunk 1]
```diff
@@ -59,6 +61,34 @@ ColumnPtr ColumnFunction::cut(size_t start, size_t length) const
...
 }
 
+void ColumnFunction::insertFrom(const IColumn & src, size_t n)
+{
+    const ColumnFunction & src_func = assert_cast<const ColumnFunction &>(src);
+
+    for (size_t i = 0, size = captured_columns.size(); i < size; ++i)
+    {
+        auto mut_column = IColumn::mutate(std::move(captured_columns[i].column));
+        mut_column->insertFrom(*src_func.captured_columns[i].column, n);
```
Comment (kitaisreal) [inline src/Columns/ColumnFunction.cpp:71]: Should not we check that src_func.captured_columns size equals to our captured_columns size ?

Comment (kitaisreal) [reply src/Columns/ColumnFunction.cpp:71]: Also this function can just use `insertRangeFrom` with start = n and length = 1.

**Comment by kitaisreal** [comment]
Maybe add a test for such scenario ?

**Comment by alexey-milovidov** [comment]
Thank you! I forgot to commit the test that I've did.

File: src/Columns/ColumnFunction.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply src/Columns/ColumnFunction.cpp:71]: > Should not we check that src_func.captured_columns size equals to our captured_columns size ?

Yes, I will add assert.

> Also this function can just use insertRangeFrom with start = n and length = 1.

`insertRangeFrom` can be more heavy for some columns, so I've implemented a separate method.

**Comment by alexey-milovidov** [comment]
I've added a test and made the changes.

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**